src/gamepuzzles/wordPuzzleLogic.ts

```diff
@@ -0,0 +1,146 @@
+export default class WordPuzzleLogic {
+    /**
+        puzzleNumber - Puzzle stage level of current (Up to 5 stage levels) game level.
+        groupedLetters - String sequence of letters when performing the multi-letter seleciton.
+        droppedLetters - String sequence of letters when group of letters was fed to the monster.
+        groupedObj - Object with key properties of stone letter index, used for validating duplicate letters while hovering.
+        droppedHistory - Object with key properties of stone letter index that was fed to the monster.
+                        Used to preserve the list for hiding the stone letters.
+        hideListObj - Object with key properties of stone letter index.
+                    Used to hide stones that is part of the group or stones that was already fed to the monster.
+    **/
+    private levelData: {
+        levelNumber: number;
+        levelMeta: {
+            letterGroup: number;
+            levelNumber: number;
+            levelType: string;
+            promptFadeOut: number;
+            protoType: string;
+        };
+        puzzles: [
+            {
+                foilStones: string[];
+                prompt: {
+                    promptAudio: string;
+                    promptText: string;
+                };
+                segmentNumber: number;
+                targetStones: string[];
+            }
+        ];
+    }
+    private puzzleNumber: number;
+    private groupedLetters: string;
+    private droppedLetters: string;
+    private groupedObj: {} | { [key:number]: string };
+    private droppedHistory: {} | { [key:number]: string };
+    private hideListObj: {} | { [key:number]: string };
+
+    constructor(levelData, puzzleNumber) {
+        this.levelData = levelData;
+        this.puzzleNumber = puzzleNumber;
+        this.groupedLetters = '';
+        this.droppedLetters = '';
+        this.groupedObj = {};
+        this.droppedHistory = {};
+        this.hideListObj = {};
+    }
+
+    private getTargetWord():string {
+        return this.levelData.puzzles[this.puzzleNumber]?.prompt?.promptText;
+    }
+
+    getValues(): {
+        groupedLetters: string;
+        droppedLetters: string;
+        groupedObj: { [key:number]: string },
+        droppedHistory: { [key:number]: string },
+        hideListObj: { [key:number]: string },
+    } {
+        return {
+            groupedLetters: `${this.groupedLetters}`,
+            droppedLetters: `${this.droppedLetters}`,
+            groupedObj: { ...this.groupedObj },
+            droppedHistory: { ...this.droppedHistory },
+            hideListObj: { ...this.hideListObj },
+        }
+    }
+
+    checkIsWordPuzzle():boolean {
+        return this.levelData?.levelMeta?.levelType === 'Word';
+    }
+
+    updatePuzzleLevel(puzzleNumber: number):void {
+        this.clearAllValues();
+        this.puzzleNumber = puzzleNumber;
+    }
+
+    clearPickedUp():void {
+        this.groupedLetters = '';
```

Comment:
Avoid Repeating Initialization Code in Constructor and Methods. Consider using a separate method for resetting values to avoid duplication